File: 10530 Guessing Game/solution.c
#include <stdio.h>
#include <string.h>

int main() {
	int min = 1, max = 10, guess;
	char resp[10];
	while(scanf("%d", &guess) && guess != 0) {
	    getchar();
	    fgets(resp, sizeof(resp), stdin);
	    switch(resp[4]) {
	        case 'h':
	            max = max < (guess - 1) ? max : guess - 1;
	            break;
	        case 'l':
	            min = min > (guess + 1) ? min : guess + 1;
	            break;
	        case 't':
	            if(guess >= min && guess <= max)
    	            puts("Stan may be honest");
    	        else
    	            puts("Stan is dishonest");
    	        min = 1;
    	        max = 10;
	    }
	}
	return 0;
}

File: 10530 Guessing Game/solution.java
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
		int min = 1, max = 10;
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StreamTokenizer st = new StreamTokenizer(br);
		int guess;
		StringBuilder output = new StringBuilder();
		while(st.nextToken() == StreamTokenizer.TT_NUMBER && (guess = (int)st.nval) != 0) {
		    String resp = br.readLine();
		    if(resp.equals("too high"))
                max = Math.min(max, guess - 1);
            else if(resp.equals("too low"))
                min = Math.max(min, guess + 1);
            else if(resp.equals("right on")) {
                if(guess >= min && guess <= max)
                    output.append("Stan may be honest\n");
                else
                    output.append("Stan is dishonest\n");
                min = 1;
                max = 10;
            }
		}
		System.out.print(output);
	}
}
